crates/ruff_linter/src/rules/flake8_annotations/rules/definition.rs

```diff
@@ -915,8 +912,8 @@ pub(crate) fn definition(
                 .or_else(|| parameters.args.first())
                 .is_some_and(|first_param| first_param.annotation().is_some()))
     {
-        diagnostics
-    } else {
-        vec![]
+        for diagnostic in diagnostics {
+            checker.report_diagnostic(diagnostic);
+        }
```

Comment:
This will be interesting to replace but I suspect will have to make this entire check earlier (or use an internal intermediate representation for the diagnostics, or have a separate `report_diagnostic` API for cases where diagnostics are created first and then maybe dropped later, sort of a defuse method)